Sources/LLVM/ORCJIT.swift

```diff
@@ -0,0 +1,247 @@
+#if !NO_SWIFTPM
+import cllvm
+#endif
+
+/// A SymbolResolver is an object that has special logic for looking up symbols
+/// in the JIT.
+public protocol SymbolResolver {
+  /// Determines the callable address of the provided function in the JIT's
+  /// address space.
+  ///
+  /// - Parameters:
+  ///   - symbol: The name of the symbol to look up.
+  ///   - jit: The JIT in which the resolver is looking.
+  /// - Returns: `nil` if the symbol could not be found, otherwise the opaque
+  ///            address of a C function pointer that can be called to execute
+  ///            the compiled code.
+  func address(of symbol: String, in jit: ORCJIT) -> UInt64?
+}
+
+
+/// A SimpleSymbolResolver looks through the JIT's symbol table to find
+/// addresses.
+public struct SimpleSymbolResolver: SymbolResolver {
+  public func address(of symbol: String, in jit: ORCJIT) -> UInt64? {
+    var target: LLVMOrcTargetAddress = 0
+    let err = LLVMOrcGetSymbolAddress(jit.llvm, &target, symbol)
+    guard err == LLVMOrcErrSuccess else {
+      return nil
+    }
+    return target
+  }
+}
+
+/// Runs a series of symbol resolvers in order, stopping when the first resolver
+/// finds a valid symbol.
+public struct CompoundSymbolResolver: SymbolResolver {
```

Comment:
@CodaFi should compound resolvers just...be the default? i.e. should ORCJIT always keep a chain of resolvers and run them, defaulting to the simple resolver?

Reply:
Ooh, neat.  Sure, sounds good.  Then symbol resolution is a fold with ??.